exporter/datadogexporter/internal/logs/translator.go

```diff
@@ -0,0 +1,168 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logs // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/logs"
+
+import (
+	"encoding/binary"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/otlp/model/attributes"
+	"github.com/DataDog/datadog-agent/pkg/otlp/model/source"
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadog"
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+)
+
+const (
+	namespace          = "otel"
+	otelTraceID        = namespace + ".trace_id"
+	otelSpanID         = namespace + ".span_id"
+	otelSeverityNumber = namespace + ".severity_number"
+	otelSeverityText   = namespace + ".severity_text"
+	otelTimestamp      = namespace + ".timestamp"
+
+	ddStatus    = "status"
+	ddTraceID   = "dd.trace_id"
+	ddSpanID    = "dd.span_id"
+	ddTimestamp = "@timestamp"
+)
+
+const (
+	logLevelTrace = "trace"
+	logLevelDebug = "debug"
+	logLevelInfo  = "info"
+	logLevelWarn  = "warn"
+	logLevelError = "error"
+	logLevelFatal = "fatal"
```

Comment:
I don't think these are the official status levels we use in Datadog. I think we use the Syslog log levels defined in [this RFC](https://datatracker.ietf.org/doc/html/rfc5424) plus "OK". There's some documentation about the log levels [here](https://docs.datadoghq.com/logs/log_configuration/processors/?tab=ui&_gl=1*1soh6bj*_ga*MjEwMTYzMjkyMS4xNjYyNzMwMTgz*_ga_KN80RDFSQK*MTY2MzE1ODAzNy41LjAuMTY2MzE1ODAzNy42MC4wLjA.&_ga=2.9424181.1490963202.1663158038-2101632921.1662730183#log-status-remapperr) in the context of the status remapper feature.

Here is a screenshot of the log levels in the Datadog UI:

<img width="407" alt="Screen Shot 2022-09-14 at 2 22 53 PM" src="https://user-images.githubusercontent.com/6981620/190152986-f9ae1611-5848-4687-94e4-13b2a229e6bc.png">

We should update these to match the Datadog standard and also update the conversion function accordingly.

Reply:
This is my bad because I included the incorrect log levels in the RFC. I will add a comment to the RFC calling this out.

Reply:
That is, I am now worried about this scenario: `SeverityText: "Alert", SeverityNumber: "20"` should probably convert to `Alert` but right now it would convert to `Error`

Reply:
It looks like these log levels are still not consistent with the ones used in Datadog.

Reply:
as discussed, we are using range names as defined by  https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-severitynumber . This would be mapped to datadog log status by the backend